File: app/src/main/java/com/sunny/lib/base/BaseActivity.java
package com.sunny.lib.base;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.sunny.lib.manager.ActivityLifeManager;

public abstract class BaseActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityLifeManager.INSTANCE.onCreate(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        ActivityLifeManager.INSTANCE.onDestroy(this);

    }
}

File: app/src/main/java/com/sunny/lib/base/BaseActivity.kt
package com.sunny.lib.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.sunny.lib.manager.ActivityLifeManager
import com.sunny.lib.utils.SunToast

open class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActivityLifeManager.onCreate(this)
    }

    protected fun showToast(msg: String) {
        SunToast.show(msg)
    }

    override fun onDestroy() {
        super.onDestroy()
        ActivityLifeManager.onDestroy(this)
    }

    fun doExitActivity() {
        finish()
    }
}